.github/workflows/main.yml

```
@@ -5,23 +5,24 @@
-      - feature/code-coverage
+      - feature/CD
 
 jobs:
   build:
     name: Build and Test Application
     runs-on: ubuntu-latest
+    timeout-minutes: 30  # ⏱️ Global timeout for build job
 
     steps:
       - name: Checkout Code
         uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)